docs/powerpoint/tagging-presentations-slides-shapes.md

```diff
@@ -0,0 +1,162 @@
+
+```javascript
+async function addMultipleSlideTags() {
+  await PowerPoint.run(async function(context) {
+    const slide = context.presentation.slides.getItemAt(0);
+    slide.tags.add("OCEAN", "Arctic");
+    slide.tags.add("PLANET", "Jupiter");
+
+    await context.sync();
+  });
+}
+```
+
+The `add` method is also used to update a tag. The following example, changes the value of the `PLANET` tag.
+
+```javascript
+async function updateTag() {
+  await PowerPoint.run(async function(context) {
+    const slide = context.presentation.slides.getItemAt(0);
+    slide.tags.add("PLANET", "Mars");
+
+    await context.sync();
+  });
+}
+```
+
+To delete a tag, call the `delete` method on it's parent `TagsCollection` object and pass the key of the tag as the parameter. An example is in [Setting custom metadata on the presentation](#setting-custom-metadata-on-the-presentation).
+
+### Using tags to selectively process slides and shapes
+
+Consider the following scenario: Contoso Consulting has a presentation they show to all new customers. But some slides should only be shown to customers that have paid for "premium" status. Before showing the presentation to non-premium customers, they make a copy of it and delete the slides that only premium customers should see. An add-in enables Contoso to tag which slides are for premium customers and to delete these slides when needed. The following are the major coding steps to create this functionality.
+
+1. Create a method that tags the currently selected slide as intended for `Premium` customers. About this code, note:
+
+    - The `getSelectedSlideIndex` function is defined in the next step. It returns the 1-based index of the currently selected slide.
+    - The value returned by the `getSelectedSlideIndex` function has to be decremented because the [SlideCollection.getItemAt](/javascript/api/powerpoint/powerpoint.slidecollection#getItemAt_index_) method is 0-based.
+
+    ```javascript
+    async function addTagToSelectedSlide() {
+      await PowerPoint.run(async function(context) {
+        let selectedSlideIndex = await getSelectedSlideIndex();
+        selectedSlideIndex = selectedSlideIndex - 1;
+        const slide = context.presentation.slides.getItemAt(selectedSlideIndex);
+        slide.tags.add("CUSTOMER_TYPE", "Premium");
+    
+        await context.sync();
+      });
+    }
+    ```
+
+2. Create a method to get the index of the selected slide. The following is an example. Note about this code:
+
+    - It uses the [Office.context.document.getSelectedDataAsync](/javascript/api/office/office.document#getSelectedDataAsync_coercionType__callback_) method of the Common JavaScript APIs.
+    - The call to `getSelectedDataAsync` is embedded in a Promise-returning function. For more information about why and how to do this, see [Wrap Common APIs in promise-returning functions](../develop/asynchronous-programming-in-office-add-ins.md#wrap-common-apis-in-promise-returning-functions).
```

Comment:
In a team meeting recently, we decided: "The `Promise` object should be capitalized and in code ticks. The concept of a promise should be lower case." 
```suggestion
    - The call to `getSelectedDataAsync` is embedded in a promise-returning function. For more information about why and how to do this, see [Wrap Common APIs in promise-returning functions](../develop/asynchronous-programming-in-office-add-ins.md#wrap-common-apis-in-promise-returning-functions).
```